docs/assembly.rst

```diff
@@ -96,6 +96,31 @@ you really know what you are doing.
                 }
             }
         }
+
+        // Same as above, but accomplish the entire code within inline assembly.
+        function sumPureAsm(uint[] _data) returns (uint o_sum) {
+            assembly {
+               // Load the length (first 32 bytes)
+               let len := mload(_data)
+
+               // Skip over the length field.
+               _data := add(_data, 0x20)
+
+               // Iterate until the bound is not met.
+               for
+               {
+                   let end := add(_data, len)
+               }
+               lt(_data, end)
+               {
+                   _data := add(_data, 0x20)
+               }
+               {
+                   o_sum := add(o_sum, mload(_data))
+                   _data := add(_data, 0x20)
```

Comment:
This line should be removed now.
Can you fit the header part into a single line? If not, I agree that it is not too readable.